Assignment4/s100257_SPARQL(Basic).txt

```diff
@@ -0,0 +1,311 @@
+	?Instance ?Label
+WHERE {
+	?Instance a onto:City.
+	?Instance rdfs:label ?Label.
+}
+
++-----------------------------------------------------------+-------------------------+
+|	Instance                                                  |	Label                   |
++-----------------------------------------------------------+-------------------------+
+|	http://GP-onto.fi.upm.es/exercise2#ACoruC1a               |	A Coruña                |
+|	http://GP-onto.fi.upm.es/exercise2#Santiago_de_Compostela |	Santiago_de_Compostela  |
+|	http://GP-onto.fi.upm.es/exercise2#Madrid                 |	Madrid                  |
++-----------------------------------------------------------+-------------------------+
+
+
+4. Get the number of inhabitants of Santiago de Compostela
+
+PREFIX onto: <http://GP-onto.fi.upm.es/exercise2#>
+
+SELECT DISTINCT
+	?InhabitantNumber
+WHERE {
+	onto:Santiago_de_Compostela onto:hasInhabitantNumber ?InhabitantNumber.
+}
+
++---------------------------------------------------------+
+|	InhabitantNumber                                        |
++---------------------------------------------------------+
+|	" 300000 "^^<http://www.w3.org/2001/XMLSchema#integer>  |
++---------------------------------------------------------+
+
+
+5. Get the number of inhabitants of Santiago de Compostela and Arzua
+
+PREFIX onto: <http://GP-onto.fi.upm.es/exercise2#>
+
+SELECT DISTINCT
+	?Place ?Label ?InhabitantNumber
+WHERE {
+	{
+		?Place onto:hasInhabitantNumber ?InhabitantNumber.
+		filter (?Place = onto:Santiago_de_Compostela).
+	}
+		union
+	{
+		?Place onto:hasInhabitantNumber ?InhabitantNumber.
+		filter (?Place = onto:Arzua).
+	}
+	?Place rdfs:label ?Label.
+}
+
++-----------------------------------------------------------+-------------------------+---------------------------------------------------------+
+|	Place                                                     |	Label                   |	InhabitantNumber                                        |
++---------------------------------------------------------------+---------------------------+---------------------------------------------------+
+|	http://GP-onto.fi.upm.es/exercise2#Santiago_de_Compostela |	Santiago_de_Compostela  |	" 300000 "^^<http://www.w3.org/2001/XMLSchema#integer>  |
+|	http://GP-onto.fi.upm.es/exercise2#Arzua                  |	Arzua                   |	" 38900 "^^<http://www.w3.org/2001/XMLSchema#integer>   |
++-----------------------------------------------------------+-------------------------+---------------------------------------------------------+
+
+
+6. Get all places, together with the number of inhabitants, ordered by the place name (ascending)
+
+PREFIX onto: <http://GP-onto.fi.upm.es/exercise2#>
+
+SELECT DISTINCT
+	?Place ?Label ?InhabitantNumber
+WHERE {
+	?Place onto:hasInhabitantNumber ?InhabitantNumber.
+	?Place a ?Locality.
+	?Locality rdfs:subClassOf onto:Locality.
+	?Place rdfs:label ?Label.
+}
+ORDER BY
+	?Place ASC(?InhabitantNumber)
```

Comment:
You are ordering wrongly since the inhabitants are written as texts and not as numbers. In any case, the query is ok.